packages/core/test/unit/application.unit.ts

```diff
@@ -35,9 +42,26 @@ describe('Application', () => {
 
   describe('component binding', () => {
     let app: Application;
+    const binding = new Binding('foo');
     class MyController {}
+    class MyClass {}
+    class MyProvider implements Provider<string> {
+      value() {
+        return 'my-str';
+      }
+    }
     class MyComponent implements Component {
       controllers = [MyController];
+      bindings = [binding];
+      classes = {'my-class': MyClass};
+      providers = {'my-provider': MyProvider};
```

Comment:
Instead of treating `MyComponent` as a sort of a kitchen sink, please create a minimal component class in each test case.

For example:

```ts
    it('binds providers from a component', () => {
       class ComponentWithProvider implements Component {
         providers = {'my-provider': MyProvider};
       }

       app.component(ComponentWithProvider);

       expect(app.contains('my-provider')).to.be.true();
       expect(app.getSync('my-provider')).to.be.eql('my-str');
     });
```

My proposed design makes the tests much easier to understand and modify.